pennylane/devices/qubit/apply_operation.py

```diff
@@ -349,82 +350,82 @@ def apply_mid_measure(
     mid_measurements = execution_kwargs.get("mid_measurements", None)
     rng = execution_kwargs.get("rng", None)
     prng_key = execution_kwargs.get("prng_key", None)
-    postselect_mode = execution_kwargs.get("postselect_mode", None)
 
     if is_state_batched:
         raise ValueError("MidMeasureMP cannot be applied to batched states.")
+
     wire = op.wires
-    interface = qml.math.get_deep_interface(state)
+    interface = math.get_deep_interface(state)
 
-    if postselect_mode == "fill-shots" and op.postselect is not None:
-        sample = op.postselect
-    else:
-        axis = wire.toarray()[0]
-        slices = [slice(None)] * qml.math.ndim(state)
-        slices[axis] = 0
-        prob0 = qml.math.real(qml.math.norm(state[tuple(slices)])) ** 2
+    axis = wire.toarray()[0]
+    slices = [slice(None)] * math.ndim(state)
+    slices[axis] = 0
+    prob0 = math.real(math.norm(state[tuple(slices)])) ** 2
 
-        if prng_key is not None:
-            # pylint: disable=import-outside-toplevel
-            from jax.random import binomial
+    if prng_key is not None:
+        # pylint: disable=import-outside-toplevel
+        from jax.random import binomial
 
-            def binomial_fn(n, p):
-                return binomial(prng_key, n, p).astype(int)
+        def binomial_fn(n, p):
+            return binomial(prng_key, n, p).astype(int)
 
-        else:
-            binomial_fn = np.random.binomial if rng is None else rng.binomial
-        sample = binomial_fn(1, 1 - prob0)
+    else:
+        binomial_fn = np.random.binomial if rng is None else rng.binomial
+
+    sample = binomial_fn(1, 1 - prob0)
```

Comment:
This is the only section that matters, everything else in this file is cleaning up imports.